### PR TITLE
[TSDK-296] Support collective and group events

### PR DESCRIPTION
**Pull request by mrashed-dev**
# Description
This PR bring support to [collective](https://developer.nylas.com/docs/connectivity/calendar/calendar-availability/#collective-meetings) and [group events](https://developer.nylas.com/docs/connectivity/calendar/calendar-availability/#group-meetings).

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: src/models/event.ts [hunk 1]
```diff
@@ -329,4 +336,26 @@ export default class Event extends RestfulModel {
...
+    ) {
+      throw new Error(
+        "Cannot set both 'details' and 'autocreate' in conferencing object."
+      );
+    }
+    if (
+      this.capacity &&
+      this.capacity != -1 &&
+      this.participants &&
+      this.participants.length <= this.capacity
```
Comment (ozsivanov) [inline src/models/event.ts:354]: I don't think participants are included in the event capacity, since they're part of the meeting by default. cc @tbadun

Comment (tbadun) [reply src/models/event.ts:354]: ^ that's just a weird internal verbiage thing, the participants list in an event is the sum total of organizers, participants, attendees, and invitees

Comment (tbadun) [reply src/models/event.ts:354]: our cc-side validation looks like:
```python
def validate_capacity_request(self, capacity):
    # type: (int) -> None
    if (not isinstance(capacity, int)) or (
        (capacity == 0) or (capacity > 1000) or (capacity < -1)
    ):
        raise InputError(
            "Allowed capacity is between 1-1000 or -1 for unlimited: {}".format(
                capacity
            )
        )
    # only count none-organizer
    num_participants = len(
        [
            participant
            for participant in self.participants
            if participant.get("role") != "organizer"
        ]
    )

    if 0 < capacity < num_participants:
        raise InputError("Participants over capacity: {}".format(capacity))
```

Comment (ozsivanov) [reply src/models/event.ts:354]: Cool, just wanted to double check 👍

**Review by ozsivanov — APPROVED** [review]
LGTM!

File: src/models/event.ts (hunk 1 above)
Comment (tbadun) [reply src/models/event.ts:354]: "participants" means nothing and everything at the same time, just to make communication around scheduler the most fun possible

Comment (mrashed-dev) [reply src/models/event.ts:354]: @ozsivanov @tbadun thank you both!

**Pull request merged**